app/controllers/registrations_controller.rb

```diff
@@ -8,4 +8,19 @@ def new
       super
     end
   end
+
+  def create
+    not_authorized unless SiteConfig.allow_email_password_registration
+
+    build_resource(sign_up_params)
+    resource.saw_onboarding = false
+    resource.editor_version = "v2"
+    resource.save if resource.email.present?
```

Comment:
Would it make sense to add a check for the number of registered users and if it happens to be the first user in the instance maybe make the user admin?

This may be too much of a hack so take it a grain of salt (to say the least). It's just an idea to streamline the process in this beta period for creators.

Reply:
> Would it make sense to add a check for the number of registered users and if it happens to be the first user in the instance maybe make the user admin?

This is approximately how I'm thinking about it, or just have forems live in a `SiteConfig.setup_mode = true` or something like that.

This pass doesn't add that specific "should I make this person a super admin logic", I'll add that after we get this in.

Reply:
Cool! Following up on that concept of setup mode: We already have the `setup_completed?` helper, which "means" the instance hasn't been configured yet (requires an admin) which could save us adding another SiteConfig.

Anyways, just sharing ideas here because of course doing that over a SiteConfig has its own set of implications.